src/services/refactors/extractSymbol.ts

```diff
@@ -1369,7 +1368,28 @@ function extractConstantInScope(
 
     const changeTracker = textChanges.ChangeTracker.fromContext(context);
 
-    if (isClassLike(scope)) {
+    if (isArrowFunction(scope) && !isFunctionBody(scope.body)) {
+        const localReference: Expression = factory.createIdentifier(localNameText);
+
+        const newVariableDeclaration = factory.createVariableDeclaration(localNameText, /*exclamationToken*/ undefined, variableType, getSynthesizedDeepClone(initializer));
+        const newVariableStatement = factory.createVariableStatement(
+            /*modifiers*/ undefined,
+            factory.createVariableDeclarationList([newVariableDeclaration], NodeFlags.Const));
+
+        const returnExpression = visitNode(
+            scope.body,
+            function visitor(n: Node): VisitResult<Node> {
+                if (n === node) {
+                    return localReference;

```

Comment:
I think that I should also copy comments from `n` to `localReference` here but [issues like this](https://github.com/microsoft/TypeScript/issues/55348) makes me believe that this might be slightly more complicated than just calling `copyComments(n, localReference)`. Any tips?